### PR TITLE
Update reportengine dependencies

### PR DESCRIPTION
**Pull request by scarlehoff**
This PR does:

- updates the `ruamel` and `dask` dependencies 

- Modify tests in `test_executor.py` and `test_builder.py` so as to be able to accept dask futures.

### CODE REVIEW

**Comment by scarlehoff** [comment]
I will merge this by the time https://github.com/NNPDF/nnpdf/pull/1861 is implemented.

If someone (@comane I guess is the only person with experience here) can review before that so much the better.

Since this is only dealing with a few test and dependencies and the latest version is already being used via conda, I guess it would be fine.

**Comment by RoyStegeman** [comment]
@comane when you have time, could you please review this?

File: src/reportengine/tests/test_executor.py [hunk 1]
```diff
@@ -76,10 +76,13 @@ def nsspec(x, beginning=()):
         self.graph.add_node(mcall, inputs={gcall, hcall})
 
 
-    def _test_ns(self):
+    def _test_ns(self, promise=False):
```
Comment (comane) [inline src/reportengine/tests/test_executor.py:79]: Is there a particular reason for calling this `promise`?

**Review by comane — APPROVED** [review]
Hi @scarlehoff, @RoyStegeman, I simply updated the description of the PR as this is not only updating the dependencies

**Comment by RoyStegeman** [comment]
@scarlehoff could you add a comment explaining why you restricted the `ruamel_yaml` version?

File: pyproject.toml [hunk 2]
```diff
@@ -19,12 +19,12 @@ classifiers = [
 description-file="README.md"
 requires = [
     "jinja2",
-    "ruamel_yaml",
+    "ruamel_yaml<0.18",
```
Comment (scarlehoff) [inline pyproject.toml:22]:
```suggestion
    "ruamel_yaml<0.18", # the code is not compatible with ruamel 0.18
```

File: src/reportengine/tests/test_executor.py (hunk 1 above)
Comment (scarlehoff) [reply src/reportengine/tests/test_executor.py:79]: I think it looked like a js promise objet to me

**Comment by scarlehoff** [comment]
@comane are you fine with me merging this?

afaik you are the only one using the newer versions of validphys

File: pyproject.toml [hunk 3]
```diff
@@ -19,12 +19,12 @@ classifiers = [
...
     "jinja2",
-    "ruamel_yaml",
+    "ruamel_yaml<0.18", # the code is not compatible with ruamel 0.18
     "matplotlib",
     "pandas",
     "pygments",
     "blessings",
-    "dask",
+    "dask[distributed]",
 ]
```
Comment (comane) [inline pyproject.toml:28]:
```suggestion
    "dask[distributed]",
    "bokeh!=3.0.*,>=2.4.2",
]
```

Comment (scarlehoff) [reply pyproject.toml:28]: is bokeh a required dependency? I don't see it anywhere

**Comment by comane** [comment]
Hi @scarlehoff, I tested this branch with the most up to date version of validphys (nnpdf), by pip installing this branch.
I runned some example scripts (plot pdfs and data theory comparison) and they worked well even with dask.

I guess that the idea is to change the nnpdf dependency as well since right now it is checking a fixed commit (which also needs curio).


I added a suggestion for bokeh. This way one gets the dask dashboard.

**Comment by scarlehoff** [comment]
I see. I'll add bokeh then as an extra. Since dependencies of reportengine affect nnpdf as well I'd rather not add many.

> I guess that the idea is to change the nnpdf dependency as well since right now it is checking a fixed commit (which also needs curio).

Yes. At least for python 3.12.
But the first step is to create a conda package for this (since that's the blocker right now)

**Pull request merged**